### PR TITLE
Network Quality fixes 1

### PR DESCRIPTION
**Pull request by lakshyacis**
**Changes based on flake8 and pylint.**
Reduce line length| Suppressed warnings| Added docstrings.

### CODE REVIEW

File: src/network/__init__.py [hunk 1]
```diff
@@ -1,3 +1,7 @@
+"""
+Networking package. Helps to connects with the network
```
Comment (PeterSurda) [inline src/network/__init__.py]: Network subsystem package

File: src/network/addrthread.py [hunk 2]
```diff
@@ -1,3 +1,6 @@
+"""
+Adds the threads
```
Comment (PeterSurda) [inline src/network/addrthread.py]: Address broadcasting thread

File: src/network/addrthread.py [hunk 3]
```diff
@@ -7,17 +10,19 @@
 
 
 class AddrThread(StoppableThread):
+    """"""
```
Comment (PeterSurda) [inline src/network/addrthread.py]: ´"""(Node) address broadcasting thread"""`

File: src/network/connectionchooser.py [hunk 4]
```diff
@@ -1,3 +1,7 @@
+"""
+Connects with the appropriate connection
```
Comment (PeterSurda) [inline src/network/connectionchooser.py]: Select which node to connect to.

File: src/network/connectionchooser.py [hunk 5]
```diff
@@ -12,6 +16,7 @@
 
 
 def getDiscoveredPeer():
+    """"""
```
Comment (PeterSurda) [inline src/network/connectionchooser.py]: `"""Get a peer from the local peer discovery list"""`

File: src/network/networkthread.py [hunk 6]
```diff
@@ -1,3 +1,7 @@
+"""
+Handle's network related tasks
```
Comment (PeterSurda) [inline src/network/networkthread.py]: Main network thread

File: src/network/receivequeuethread.py [hunk 7]
```diff
@@ -1,3 +1,6 @@
+"""
+Threading activities
```
Comment (PeterSurda) [inline src/network/receivequeuethread.py]: Process data incoming from network

File: src/network/bmproto.py [hunk 8]
```diff
@@ -318,8 +318,8 @@ def decode_simple(self, char="v"):  # pylint: disable=inconsistent-return-statem
 
     def bm_command_error(self):
         """Decode an error message and log it"""
-        fatalStatus, banTime, inventoryVector, errorText = \
-            self.decode_payload_content("vvlsls")
+        fatalStatus, errorText = \
+            self.decode_payload_content("vvlsls")[0], self.decode_payload_content("vvlsls")[3]
```
Comment (PeterSurda) [inline src/network/bmproto.py]: Perhaps I would add a comment describing elements 1 and 2 for documentation purposes, as the error command isn't documented well.

**Comment by PeterSurda** [comment]
please remove `bitmessagecureses/__init__py` from the PR

File: src/network/bmproto.py (hunk 8 above)
Comment (g1itch) [reply src/network/bmproto.py]: this change has few sense and may cause problems, better:
```python
        err_values = self.decode_payload_content("vvlsls")
        fatalStatus = err_values[0]
        errorText = err_values[3]
```

File: src/network/addrthread.py [hunk 9]
```diff
@@ -7,17 +11,19 @@
...
+        """Method representing the threads activity."""
         while not state.shutdown:
             chunk = []
             while True:
                 try:
                     data = addrQueue.get(False)
                     chunk.append((data[0], data[1]))
                     if len(data) > 2:
-                        source = BMConnectionPool().getConnectionByAddr(data[2])
+                        source = BMConnectionPool().getConnectionByAddr(data[2])  # pylint: disable=unused-variable
```
Comment (g1itch) [inline src/network/addrthread.py]: len 115

File: src/network/addrthread.py [hunk 10]
```diff
@@ -7,17 +11,19 @@
 
 
 class AddrThread(StoppableThread):
+    """(Node) address broadcasting thread"""
     name = "AddrBroadcaster"
 
     def run(self):
+        """Method representing the threads activity."""
```
Comment (g1itch) [inline src/network/addrthread.py]: Hello, cap. This docstring describes nothing.

File: src/network/receivequeuethread.py [hunk 11]
```diff
@@ -10,6 +14,7 @@
 
 
 class ReceiveQueueThread(StoppableThread):
+    """Represents thread activities"""
```
Comment (g1itch) [inline src/network/receivequeuethread.py]: No, it's class definition.

File: src/network/addrthread.py [hunk 12]
```diff
@@ -1,3 +1,7 @@
+"""
+(Node) address broadcasting thread
+"""
```
Comment (g1itch) [inline src/network/addrthread.py]: AddrThread class definition

File: src/network/networkthread.py [hunk 13]
```diff
@@ -1,3 +1,7 @@
+"""
+Main network thread
+"""
```
Comment (g1itch) [inline src/network/networkthread.py]: This docstring better fits the class definition, not module.

File: src/network/addrthread.py (hunk 9 above)
Comment (g1itch) [reply src/network/addrthread.py]: if it's unused, don't define it

File: src/network/bmproto.py [hunk 14]
```diff
@@ -436,7 +436,7 @@ def bm_command_addr(self):
         """Incoming addresses, process them"""
         addresses = self._decode_addr()      # pylint: disable=redefined-outer-name
         for i in addresses:
-            seenTime, stream, services, ip, port = i
+            seenTime, stream, _, ip, port = i
```
Comment (g1itch) [inline src/network/bmproto.py]:
```python
for seenTime, stream, services, ip, port in addresses:
    ...
```

**Comment by g1itch** [comment]
So inconsistent. I don't understand why do you
 - edit plugins again;
 - document `run()` only in `AddrThread`;
 - edit unrelated nodules in single PR.

**Comment by lakshyacis** [comment]
> So inconsistent. I don't understand why do you
> 
> * edit plugins again;
> * document `run()` only in `AddrThread`;
> * edit unrelated modules in single PR.

Actually, all the code quality stuff is done. So, I'm just looking for the files that needed fixes. That's why I edited plugins again and this PR contains unrelated modules. But I will remove unrelated modules from the PR and make a new PR for them. 
Thank you

**Comment by PeterSurda** [comment]
remove addrthread.py from the PR, see #1575

**Comment by lakshyacis** [comment]
> remove addrthread.py from the PR, see #1575

Okay.

File: src/network/announcethread.py [hunk 15]
```diff
@@ -1,8 +1,6 @@
 """
-src/network/announcethread.py
-=================================
+Manages Nodes
```
Comment (PeterSurda) [inline src/network/announcethread.py]: Announce myself (node address)

File: src/network/__init__.py [hunk 16]
```diff
@@ -1,3 +1,6 @@
+"""
+Network subsystem packages
```
Comment (g1itch) [inline src/network/__init__.py]: Why plural? `pybitmessage.network` is a package

File: src/network/assemble.py [hunk 17]
```diff
@@ -13,20 +12,20 @@
 def assemble_addr(peerList):
     """Create address command"""
     if isinstance(peerList, Peer):
-        peerList = (peerList)
+        peerList = peerList
```
Comment (g1itch) [inline src/network/assemble.py]: `peerList = [peerList]`

File: src/network/asyncore_pollchoose.py [hunk 18]
```diff
@@ -50,16 +45,17 @@
 many of the difficult problems for you, making the task of building
 sophisticated high-performance network servers and clients a snap.
 """
```
Comment (g1itch) [inline src/network/asyncore_pollchoose.py]: Why do you want to include license into the doc?

File: src/network/announcethread.py (hunk 15 above)
Comment (lakshyacis) [reply src/network/announcethread.py]: Thanks

File: src/network/__init__.py (hunk 16 above)
Comment (lakshyacis) [reply src/network/__init__.py]: Yes, you're right. My bad. Thank you

File: src/network/asyncore_pollchoose.py (hunk 18 above)
Comment (lakshyacis) [reply src/network/asyncore_pollchoose.py]: I don't know about this, that's why I didn't remove. Should I remove that?

File: src/network/bmproto.py [hunk 19]
```diff
@@ -102,7 +102,7 @@ def state_bm_header(self):
         return True
 
     def state_bm_command(self):     # pylint: disable=too-many-branches
-        """Process incoming command"""
+        """Expects the presence of command"""
```
Comment (PeterSurda) [inline src/network/bmproto.py]: I think the original docstring is better

File: src/network/bmproto.py [hunk 20]
```diff
@@ -81,7 +80,8 @@ def bm_proto_reset(self):
         self.object = None
 
     def state_bm_header(self):
-        """Process incoming header"""
+        """Expects the presence of header,
```
Comment (PeterSurda) [inline src/network/bmproto.py]: original docstring better

File: src/network/bmproto.py (hunk 19 above)
Comment (lakshyacis) [reply src/network/bmproto.py]: Okay . Thank you

File: LICENSE [hunk 21]
```diff
@@ -47,3 +47,28 @@ Redistribution and use in source and binary forms, with or without modification,
    2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.
 
 THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDER ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+========================================================================================================
```
Comment (PeterSurda) [inline LICENSE]: you should indicate which file this refers to, like on lines 23 and 37

**Pull request merged**